### ISSUE TITLE
Anonymous and Logged in Cart not properly merged

### ISSUE THREAD

**Issue by zenweasel** [reactioncommerce/reaction#2332]
### Expected behavior
Previous cart and items added anonymously should be combined

### Actual Behavior
The cart is combined but line items are not merged
![basic_reaction_product_and_msedge_-_win10_preview__running_](https://cloud.githubusercontent.com/assets/72819/26347542/7ac8130e-3fdc-11e7-9494-ed399ca0738d.png)

### Steps to Reproduce the Behavior
1. Login
1. Add item to cart
1. Logout
1. Add that same item to cart
1. Login as same user again
1. Observe that the two items are kept as separate line items

### Versions
```
Node: 7.10.0
NPM: 4.2.0
Meteor Node: 4.8.2
Meteor NPM: 4.5.0
Reaction CLI: 0.8.2
Reaction: 1.2.0
Reaction branch: development
Docker: 17.04.0-ce
```

**Comment by aaronjudd** [comment]
never seen this, we need to replicate, and if we are able to replicate, we need to write a test for this. (I don't think this should be happening, for sure!)

**Comment by abdulsemiu-atanda** [comment]
I've been able to reproduce this following the steps above.
<img width="1055" alt="screen shot 2017-05-25 at 10 50 47 am" src="https://cloud.githubusercontent.com/assets/10584060/26445495/54624a42-4138-11e7-8cdd-d3463c7a72ea.png">

**Comment by abdulsemiu-atanda** [comment]
Closed via #2387
